### ISSUE TITLE
Min. GPU hardware requirement for fine-tuning with XLM-R model

### ISSUE THREAD

**Issue by WeiliangGuo** [ThilinaRajapakse/simpletransformers#278]
Hello guys:

I attempted to fine-tune NER model with XLM-R, my server has two Nvidia 1080 Ti, 11G, but it throws an OOM error no matter how small the batch size is.

So if it's caused by my current GPU limitations, then what is the minimum GPU requirement  for making it work?

Many thanks :)

**Comment by ThilinaRajapakse** [comment]
11 GB should be sufficient for XLM-R base model at the very least (probably up to a batch size of 8). Can you check GPU memory usage and see whether the python process actually consumes all 11 gigs?

**Comment by WeiliangGuo** [comment]
> 11 GB should be sufficient for XLM-R base model at the very least (probably up to a batch size of 8). Can you check GPU memory usage and see whether the python process actually consumes all 11 gigs?

Thanks for reply. Your package is quite handy. :) 
A trivial issue I found: it seems that the multi-processing feature doesn't work with Flask and FastAPI, the API will get stuck when doing predictions. Though Tornado seems OK with this.

I was dealing with XLM-R large, not tried base version yet. I might try the base soon to see if it can beat BERT multilingual base (The dataset is multilingual).
The error message was something like "tried allocate xxx MB, but XXGB was already reserved by ..."
I tried run XLM-R large on my Macbook Pro without GPU (undoubtedly slow) for a text classification task and it indeed shows better performance than using BERT multilingual base.

So I'd like to know the GPU requirement for large XLM-R large :)

**Comment by ThilinaRajapakse** [comment]
Multiprocessing can be turned off if necessary. I wouldn't recommend it though, converting to features can take several hours with large datasets when multiprocessing isn't used.

You can get a rough idea of the memory requirement by looking at how much RAM is consumed on your Macbook when the model is being trained.

I don't have the XLM-R large model downloaded atm, but I might be able to check it later.

**Comment by WeiliangGuo** [comment]
> Multiprocessing can be turned off if necessary. I wouldn't recommend it though, converting to features can take several hours with large datasets when multiprocessing isn't used.
> 
> You can get a rough idea of the memory requirement by looking at how much RAM is consumed on your Macbook when the model is being trained.
> 
> I don't have the XLM-R large model downloaded atm, but I might be able to check it later.

For predicting during feature conversion for 0.1 m+ sentences with moderate text length, It took over 10 minutes by using 2 1080 Ti, however it seemed the VRAM didn't be throughly consumed (one GPU was used around 30% of 11GB, while another's was barely consumed) even n_gpu set to 2, and multi-processing set True. Anything else I can do to better utilize current GPU VRAM to speed it up?

**Comment by ThilinaRajapakse** [comment]
Multiprocessing is for the feature conversion before training actually starts. You can increase the `train_batch_size` until the GPU memory is fully utilized.

**Comment by WeiliangGuo** [comment]
> Multiprocessing is for the feature conversion before training actually starts. You can increase the `train_batch_size` until the GPU memory is fully utilized.

Last message actually I was talking about prediction/ evaluation, not training. 

I just noticed gradient_accumulation_steps can be set, will it be significantly helpful overcoming memory limit if I set it with a large value? (I will try it soon)
 It doesn't matter for me to sacrifice more time if large models with larger batch size can be trained.
